larky/src/main/resources/vendor/Crypto/OpenPGP/__init__.star

```diff
@@ -0,0 +1,265 @@
+        """
+
+        chunk = _ensure_bytes(2, chunk, g)
+        tag = ord(chunk[0:1]) & 63
+        length = ord(chunk[1:2])
+
+        if length < 192: # One octet length
+            if len(chunk) > 2:
+                g.push(chunk[2:])
+            return (tag, length)
+        if length > 191 and length < 224: # Two octet length
+            chunk = _ensure_bytes(3, chunk, g)
+            if len(chunk) > 2:
+                g.push(chunk[3:])
+            return (tag, ((length - 192) << 8) + ord(chunk[2:3]) + 192)
+        if length == 255: # Five octet length
+            chunk = _ensure_bytes(6, chunk, g)
+            if len(chunk) > 6:
+                g.push(chunk[6:])
+            return (tag, struct.unpack('!L', chunk[2:6])[0])
+        # TODO: Partial body lengths. 1 << ($len & 0x1F)
+    self.parse_new_format = parse_new_format
+
+    def parse_old_format(chunk, g):
+        """ Parses an old-format (PGP 2.6.x) OpenPGP packet.
+            http://tools.ietf.org/html/rfc4880#section-4.2.1
+        """
+        chunk = _ensure_bytes(1, chunk, g)
+        tag = ord(chunk[0:1])
+        length = tag & 3
+        tag = (tag >> 2) & 15
+        if length == 0: # The packet has a one-octet length. The header is 2 octets long.
+            head_length = 2
+            chunk = _ensure_bytes(head_length, chunk, g)
+            data_length = ord(chunk[1:2])
+        elif length == 1: # The packet has a two-octet length. The header is 3 octets long.
+            head_length = 3
+            chunk = _ensure_bytes(head_length, chunk, g)
+            data_length = struct.unpack('!H', chunk[1:3])[0]
+        elif length == 2: # The packet has a four-octet length. The header is 5 octets long.
+            head_length = 5
+            chunk = _ensure_bytes(head_length, chunk, g)
+            data_length = struct.unpack('!L', chunk[1:5])[0]
+        elif length == 3: # The packet is of indeterminate length. The header is 1 octet long.
+            head_length = 1
+            chunk = _ensure_bytes(head_length, chunk, g)
+            data_length = None
+
+        if len(chunk) > head_length:
+             g.push(chunk[head_length:])
+        return (tag, data_length)
+    self.parse_old_format = parse_old_format
+
+    def read():
+        # Will normally be overridden by subclasses
+        self.data = self.read_bytes(self.length)
+    self.read = read
+
+    def read_bytes(count):
+        chunk = _ensure_bytes(count, b'', self.input)
+        if len(chunk) > count:
+            self.input.push(chunk[count:])
+        self.length -= count
+        return chunk[:count]
+
+    return self
+
+def LiteralDataPacket(Packet):
+    """ OpenPGP Literal Data packet (tag 11).
+        http://tools.ietf.org/html/rfc4880#section-5.9
+    """
+    self = larky.mutablestruct(__name__='LiteralDataPacket', __class__=LiteralDataPacket)
+
+    def __init__(data=None, format='b', filename='data', timestamp=time()):
+        super(LiteralDataPacket, self).__init__()
```

Comment:
Super does not work in Larky. Take a look at the csv module to see how I was able to do "inheritance".

Reply:
It was not finished, updated.